packages/beacon-node/src/api/impl/beacon/state/index.ts

```diff
@@ -254,8 +254,28 @@ export function getBeaconStateApi({
       }
 
       const epoch = filters.epoch ?? computeEpochAtSlot(state.slot);
+      const stateEpoch = computeEpochAtSlot(state.slot);
+      if (filters.slot !== undefined) {
+        const epochStartSlot = computeStartSlotAtEpoch(epoch);
+        const epochEndSlot = epochStartSlot + 32 - 1;
+
+        if (filters.slot < epochStartSlot || filters.slot > epochEndSlot) {
+          throw new ApiError(400, `Slot ${filters.slot} is not in epoch ${epoch}`);
+        }
+      }
+      if (Math.abs(epoch - stateEpoch) > 1) {
+        throw new ApiError(400, `Invalid epoch value: epoch ${epoch} is out of range`);
```

Comment:
would be useful to return state epoch and explain in the error that the epoch needs to be within one epoch, "out of range" doesn't provide enough details imo